File: JavaScript/1-replace.js
'use strict';

let rx1 = /abc/g;
let s11 = 'abcdefg';
let s12 = s11.replace(rx1, '123');
console.log(s12);

let rx2 = new RegExp('abc', 'gi');
let s21 = 'abcdefgABCDEFG';
let s22 = s21.replace(rx2, '789');
console.log(s22);

File: Scala/1-replace.scala
package regexp

object Replace extends App {
  val rx1 = "abc".r
  val s11 = "abcdefg"
  val s12 = rx1.replaceAllIn(s11, "123")
  print(s12)

  val rx2 = "(?i)abc".r
  val s21 = "abcdefgABCDEFG"
  val s22 = rx1.replaceAllIn(s11, "789")
  print(s22)
}
